File: Python/Easy/Filter/Sorting/2500_Delete_Greatest_Value_in_Each_Row.py
def deleteGreatestValue(grid: list[list[int]]) -> int:
    for row in grid:
        row.sort()
    answer = 0
    n = len(grid[0])

    for col in range(n):
        max_deleted = max(row.pop() for row in grid)
        answer += max_deleted
    return answer


grid = [[1, 2, 4], [3, 3, 1]]
print(deleteGreatestValue(grid))

File: PHP/Easy/Filters/Sorting/2500_Delete_Greatest_Value_in_Each_Row.php
<?php
function deleteGreatestValue($grid)
{
    foreach ($grid as &$row) {
        sort($row);
    }

    $answer = 0;
    $n = count($grid[0]);
    for ($col = 0; $col < $n; $col++) {
        $maxDeleted = 0;
        foreach ($grid as &$row) {
            $maxDeleted = max($maxDeleted, array_pop($row));
        }
        $answer += $maxDeleted;
    }
    return $answer;
}

$grid = [[1, 2, 4], [3, 3, 1]];
echo deleteGreatestValue($grid);